src/services/expenses.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+let expenses = [];
+
+function init() {
+  expenses = [];
+}
+
+function getAll(userId, category, from, to) {
+  let filteredExpenses = [...expenses];
+
+  if (category) {
+    filteredExpenses = filteredExpenses.filter(expense =>
+      (expense.category === category));
+  }
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(expense =>
+      (expense.userId === +userId));
+
+    return filteredExpenses;
+  }
+
+  if (from && to) {
+    filteredExpenses = filteredExpenses.filter(expense =>
+      (expense.spentAt >= from && expense.spentAt <= to));
+  }
+
+  return filteredExpenses;
+}
+
+function getById(expenseId) {
+  const foundExpense = expenses.find(expense => expense.id === +expenseId);
+
+  return foundExpense || null;
+}
+
+function create(expenseDetails) {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = expenseDetails;
+
+  const newExpense = {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+    id: expenses.length + 1,
```

Comment:
what if we delete first item from array and then add new one? id will be the same